src/utils/bv-event.class.js

```diff
@@ -53,4 +54,33 @@ class BvEvent {
   }
 }
 
+class BvModalEvent extends BvEvent {
```

Comment:
What about moving this into it's own file? Maybe even in the modal directory?

Reply:
Yeah, I wasn't sure originally... but it could be moved into the `helpers` directory inside the modal directory.

Reply:
Will need to update `modal` to use this new event anyways...

It could be inside the modal.js file as well, but a separate file would probably be better.